### PR TITLE
Add or update xml configuration for interconnects and cell distances

### PR DESCRIPTION
**Pull request by jinqi7**
Add InterconnectsXML and CellDistancesXML. Update CellCacheXML properties
 to have better usabilites.

Signed-off-by: qijing <jinqi@redhat.com>

### CODE REVIEW

**Comment by chloerh** [comment]
There are still a few code check not passed.  And it'd be much helpful if test result is provided.

File: virttest/libvirt_xml/vm_xml.py [hunk 1]
```diff
@@ -2038,25 +2167,59 @@ def marshal_to_caches(tag, new_treefile, index, libvirtxml):
         return newone
 
 
-class CellCacheXML(base.LibvirtXMLBase):
+class CellCacheXML(VMXML):
 
-    """Cell element of cache"""
```
Comment (chloerh) [inline virttest/libvirt_xml/vm_xml.py:2043]: I don't see the necessity of this change, could you help provide more specific reason?

**Review by chloerh — COMMENTED** [review]
The rest is ok to me as long as test result is good.

File: virttest/libvirt_xml/vm_xml.py (hunk 1 above)
Comment (jinqi7) [reply virttest/libvirt_xml/vm_xml.py:2043]: It's best practice from me. @kylazhang , can you please help to explain it? Thanks.

Comment (kylazhang) [reply virttest/libvirt_xml/vm_xml.py:2043]: I think the element should all based on LibvirtXMLBase so keep the origin should be right

Comment (jinqi7) [reply virttest/libvirt_xml/vm_xml.py]: After talked with kyla, "base.LibvirtXMLBase" is more better to be used. So, change to use "base.LibvirtXMLBase".  Thanks

**Comment by jinqi7** [comment]
[root@dell-per740xd-08 yum.repos.d]# avocado run --vt-type libvirt hmat
JOB ID     : e7d5d6ec23e52300ef1eec47e9a23a61279b5b13
JOB LOG    : /root/avocado/job-results/job-2020-11-09T00.04-e7d5d6e/job.log
 (1/2) type_specific.io-github-autotest-libvirt.hmat.hmat: PASS (23.49 s)
 (2/2) type_specific.io-github-autotest-libvirt.hmat.cell_distances: PASS (30.21 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 55.33 s

File: virttest/libvirt_xml/vm_xml.py [hunk 2]
```diff
@@ -2012,9 +2093,58 @@ def __init__(self, virsh_instance=base.virsh):
...
+                                     'virsh_instance': virsh_instance})
         super(NumaCellXML, self).__init__(virsh_instance=virsh_instance)
         self.xml = '<cell/>'
 
+    class CellDistancesXML(base.LibvirtXMLBase):
+        """
+        Distances of cell
+        """
+
+        __slots__ = ('sibling', 'text')
```
[the remark this answers is not in the document]
Comment (jinqi7) [reply virttest/libvirt_xml/vm_xml.py]: It's been removed. Thx

**Comment by dzhengfy** [comment]
@jinqi7 please help to rebase your branch and push your changes again to fix centos failures.

**Comment by jinqi7** [comment]
@chloerh @kylazhang , would you please help to review PR again? Thanks

**Comment by chloerh** [comment]
File `libvirt_numa.py` was created after last review.  Could you help provide more info of the newly added code?

**Review by chloerh — APPROVED** [review]

File: virttest/utils_libvirt/libvirt_numa.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    logging.debug(cpu_xml)
+    vmxml.cpu = cpu_xml
+    vmxml.sync()
+
+    return vmxml
+
+
+def create_hmat_xml(vmxml, params):
+    """
+    Create hmat xml for test
```
Comment (kylazhang) [inline virttest/utils_libvirt/libvirt_numa.py:42]: 1 blank line needed after L42

File: virttest/utils_libvirt/libvirt_numa.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    cpu_xml.numa_cell = cells
+
+    latency_list = eval(params.get('latency', ''))
+    bandwidth_list = eval(params.get('bandwidth', ''))
+    interconnects_xml = vm_xml.VMCPUXML().InterconnectsXML()
+    interconnects_xml.latency = latency_list
+    interconnects_xml.bandwidth = bandwidth_list
+
+    cpu_xml.interconnects = interconnects_xml
+    logging.debug(cpu_xml)
```
Comment (kylazhang) [inline virttest/utils_libvirt/libvirt_numa.py:74]: Please give more clear info for debug

**Review by kylazhang — COMMENTED** [review]
Others are ok for me

File: virttest/utils_libvirt/libvirt_numa.py (hunk 3 above)
Comment (jinqi7) [reply virttest/utils_libvirt/libvirt_numa.py:42]: Added

File: virttest/utils_libvirt/libvirt_numa.py (hunk 4 above)
Comment (jinqi7) [reply virttest/utils_libvirt/libvirt_numa.py]: Changed

**Review by kylazhang — APPROVED** [review]

**Pull request merged**